src/OpenTelemetry/Context/AsyncLocalDistributedContextCarrier.cs

```diff
@@ -25,7 +30,13 @@ namespace OpenTelemetry.Context
     /// </summary>
     public sealed class AsyncLocalDistributedContextCarrier : DistributedContextCarrier
     {
+#if NET452
+        // A special workaround to suppress context propagation cross AppDomains.
+        private const string ContextSlotName = "OpenTelemetry.DistributedContext";
+        private static readonly FieldInfo WrapperField = typeof(BitArray).GetField("_syncRoot", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
Q.: Why is it necessary to use a field of `BitArray` to wrap the context? Perhaps describe a bit more about the workaround.

Reply:
I've added more explanation in the comments.